app/form_builders/dsfr_form_builder.rb

```diff
@@ -0,0 +1,219 @@
+          error_message(attribute),
+          existing_file_link,
+          link_to_file(attribute),
+        ].compact
+      )
+    end
+  end
+
+  def dsfr_check_box(attribute, opts = {})
+    dsfr_input_group(attribute, opts) do
+      @template.content_tag(:div, class: "fr-checkbox-group") do
+        @template.safe_join(
+          [
+            check_box(attribute, class: input_classes(opts), disabled: check_box_disabled, **enhance_input_options(opts).except(:class)),
+            opts[:label] || label_with_hint(attribute, opts),
+          ]
+        )
+      end
+    end
+  end
+
+  def dsfr_input_group(attribute, opts, &block)
+    @template.content_tag(:div, class: input_group_classes(attribute, opts)) do
+      yield(block)
+    end
+  end
+
+  def dsfr_radio_buttons(attribute, choices, opts = {})
+    @template.content_tag(:fieldset, class: "fr-fieldset") do
+      @template.safe_join(
+        [
+          @template.content_tag(
+            :legend,
+            sanitize(@object.class.human_attribute_name(attribute).concat(hint(attribute))),
+            class: "fr-fieldset__legend--regular fr-fieldset__legend"
+          ),
+          choices.map { |choice| dsfr_radio_option(attribute, choice, opts) },
+        ]
+      )
+    end
+  end
+
+  def dsfr_radio_option(attribute, value, opts = {})
+    @template.content_tag(:div, class: "fr-fieldset__element") do
+      @template.content_tag(:div, class: "fr-radio-group") do
+        @template.safe_join(
+          [
+            radio_button(attribute, value, **opts),
+            label([attribute, value].join("_").to_sym, value: label_value(attribute)),
+          ]
+        )
+      end
+    end
+  end
+
+  def dsfr_select(attribute, choices, opts = {})
+    @template.content_tag(:div, class: "fr-select-group") do
+      @template.safe_join(
+        [
+          label_with_hint(attribute, opts),
+          dsfr_select_tag(attribute, choices, opts),
+          error_message(attribute),
+        ]
+      )
+    end
+  end
+
+  private
+
+  def dsfr_select_tag(attribute, choices, opts)
+    select(attribute, choices, { include_blank: opts[:include_blank] }, class: "fr-select", **enhance_input_options(opts).except(:class))
+  end
+
+  def dsfr_input_field(attribute, input_kind, opts = {})
+    dsfr_input_group(attribute, opts) do
+      @template.safe_join(
+        [
+          label_with_hint(attribute, opts),
+          public_send(input_kind, attribute, class: input_classes(opts), autocomplete: "off", **enhance_input_options(opts).except(:class)),
+          error_message(attribute),
+        ]
+      )
+    end
+  end
+
+  def label_with_hint(attribute, opts = {})
+    label(attribute, class: "fr-label") do
+      label_value = [label_value(attribute)]
+      label_value.push(required_tag) if opts[:required]
+
+      @template.safe_join(
+        [
+          label_value,
```

Comment:
je propose de renommer la variable `label_value` pour éviter la confusion avec le nom de méthode

```suggestion
      label_and_tags = [label_value(attribute)]
      label_and_tags.push(required_tag) if opts[:required]

      @template.safe_join(
        [
          label_and_tags,
```

Reply:
et en nitpick ultime mais sur ces lignes j’ai l’impression qu’on pourrait l’écrire avec une seule ligne 

```rb
@template.safe_join([label_and_tags, hint(opts[:hint])])
```

Reply:
et d’ailleurs d’ailleurs est-ce que ça marcherait de faire 

```rb
label_and_tags.push(hint(...))
@template.safe_join label_and_tags
```

ça n’est pas exactement la même chose en params pour safe_join mais peut-être que ça fait un rendu similaire / plus simple

Reply:
Effectivement ta dernière proposition fonctionne